cpp/test/Ice/middleware/AllTests.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) ZeroC, Inc.
+
+#include "TestHelper.h"
+#include "TestI.h"
+
+using namespace std;
+using namespace Test;
+
+class Middleware final : public Ice::Object
+{
+public:
+    Middleware(Ice::ObjectPtr next, const string& name, list<string>& inLog, list<string>& outLog)
+        : _next(std::move(next)),
+          _name(name),
```

Comment:
Pass string by value and move it?

```suggestion
    Middleware(Ice::ObjectPtr next, string name, list<string>& inLog, list<string>& outLog)
        : _next(std::move(next)),
          _name(std::move(name)),
```